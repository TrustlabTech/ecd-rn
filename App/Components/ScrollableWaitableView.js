/**
 * Early Childhood Development App
 * @copyright 2016 Global Consent Ltd
 * Civvals, 50 Seymour Street, London, England, W1H 7JG
 * @author Werner Roets <werner@io.co.za>
 */

import React, { Component } from 'react'
import {
  ScrollView,
  View,
  ActivityIndicator
} from 'react-native'
import { Colours } from '../GlobalStyles'

/**
 * A scrollable view that will display an ActivityIndicator until
 * the data is available to render the screen
 * @example
 * <ScrollableWaitableView loaded={this.state.loaded}>
 *  {this.state.apiData}
 * </ScrollableWaitableView>
 */
export default class ScrollableWaitableView extends Component {


  constructor(props) {
    super(props)
  }

  render() {
    if(this.props.loaded) {
      return (<ScrollView
                style={{flex: 1, backgroundColor: Colours.sceneBackgroundColour }}
                keyboardDismissMode="none"
                keyboardShouldPersistTaps="never"
                scrollEnable={true}
                showsVerticalScrollIndicator={false}
              >
                {this.props.children}
              </ScrollView>)
    } else {
      return (<View style={{flex: 1, alignItems: 'center', justifyContent: 'center', backgroundColor: Colours.sceneBackgroundColour}}>
                <ActivityIndicator
                  animating={true}
                  style={{height: 80}}
                  size="large"
                />
              </View>)
    }
  }
}
