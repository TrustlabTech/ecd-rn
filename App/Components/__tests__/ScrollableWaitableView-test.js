/**
 * Early Childhood Development App
 * @copyright 2016 Global Consent Ltd
 * Civvals, 50 Seymour Street, London, England, W1H 7JG
 * @author Werner Roets <werner@io.co.za>
 */

import 'react-native'
import React from 'react'
import ScrollableWaitableView from '../ScrollableWaitableView'

import renderer from 'react-test-renderer'

it('renders correctly loaded', () => {
  const tree = renderer.create(
    <ScrollableWaitableView loaded={true}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly not loaded', () => {
  const tree = renderer.create(
    <ScrollableWaitableView loaded={false}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
