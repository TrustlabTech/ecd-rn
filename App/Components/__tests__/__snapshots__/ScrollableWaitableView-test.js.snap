exports[`test renders correctly loaded 1`] = `
<ScrollView
  automaticallyAdjustContentInsets={true}
  keyboardDismissMode="interactive"
  keyboardShouldPersistTaps="handled"
  scrollEnable={true}
  showsVerticalScrollIndicator={false}>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID={undefined} />
</ScrollView>
`;

exports[`test renders correctly not loaded 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FEFEFE",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "height": 80,
      }
    } />
</View>
`;
